packages/resolve-es/src/index.js

```diff
@@ -1,54 +1,19 @@
 import 'regenerator-runtime/runtime';
-import { Readable, Writable } from 'stream';
-
-function createTransformedStream(eventStream, transforms) {
-    return transforms.reduce((eventStream, transform) => eventStream.pipe(transform), eventStream);
-}
 
 export default ({ storage, bus, transforms = [] }) => ({
-    getStreamByEventTypes(eventTypes) {
-        const eventStream = Readable({ objectMode: true, read() {} });
-        (async () => {
-            try {
-                await storage.loadEventsByTypes(eventTypes, event => eventStream.push(event));
-
-                // Custom events are allowed due ReadableStream is inherited from EventEmitter
-                // https://github.com/nodejs/node/blob/master/lib/internal/streams/legacy.js#L9
-                eventStream.emit('storageDone', true);
+    async subscribeByEventType(eventTypes, handler) {
+        await storage.loadEventsByTypes(eventTypes, handler);
```

Comment:
Very unobvious contract, the receiving end still is in waiting of the promise, and the callback function already begins to be caused - usually so don't do. If to recognize that resolve-es will be used separately from resolve-query, then such contract most likely will be inconvenient.